### ISSUE TITLE
Rain World. The Camera Problem in "The Mast"

### ISSUE THREAD

**Issue by ToxicTrash113** [snoodledev/themast#7]
### Describe the bug

 Hi! My problem occurred in "**The Mast**" Mod, for _Spearmaster_. 
 More specifically, in the region of The Mast (photos are attached). When I try to go through the hatch, the game seems to freeze - I can hear the background changing to another one, but the picture remains in the same place, the creatures continue to live their lives, I can't open the map, but I can safely go to the game menu. I encountered the first problematic hatch in photo 1. 
 Next, I tried to change the route (photo 3) and at first it even worked, but later, when restarting, it turned out that the same trouble happened to this hatch as with the first one. 
 It was possible to temporarily solve this problem by completely exiting the game and logging back in, then he could work again, but not for long, for several attempts. I tried to reach hatch No. 2 by another way, but the camera was still "hanging" on the other side. If you wait for the downpour to start, the character still won't die. I tried to disable unnecessary add-ons except for "The Mast", but this did not solve the problem.
   Text files: "BepInEx\LogOutput.log" and "RainWorld_Data\StreamingAssets\crsLog.txt " were not found.
   Please help me solve this problem
   P.S. _I have not tried to go from the left side of the "Communication Node". I will install the mod myself while already in the company for Spearmaster_

### Do you have the Downpour DLC enabled?

- [X] DLC enabled

### Attach logs

[exceptionLog.txt](https://github.com/snoodledev/themast/files/14808924/exceptionLog.txt)
[consoleLog.txt](https://github.com/snoodledev/themast/files/14808925/consoleLog.txt)

### Attach Screenshots

![2024-03-29_23-09-29](https://github.com/snoodledev/themast/assets/165425259/502c756e-2488-4231-b07b-e6e99231474f)
![2024-03-30_00-35-05](https://github.com/snoodledev/themast/assets/165425259/603a766c-d6ac-49b5-9959-72c32155596f)
![2024-03-30_00-32-07](https://github.com/snoodledev/themast/assets/165425259/a59309f0-e639-41f6-9a89-0db4ea5a6cd6)
![2024-03-30_00-37-11](https://github.com/snoodledev/themast/assets/165425259/f0c229e2-ee6a-4c37-bd2f-7e845d9b8350)

**Comment by snoodledev** [comment]
Thanks for the report! This issue seems to have been caused by some custom code being broken by the latest update, which should be fixed now. I'd recommend unsubscribing/resubscribing to the dependencies, as they've been updated recently and Steam sometimes doesn't apply that unless you resubscribe. 
Let me know how it goes 👍

**Comment by ToxicTrash113** [comment]
Thanks for the answer, I will definitely try it! One question before execution - do I have a special pearl (_a task from the Moon_), if I re-subscribe to the region, will the pearl and I transfer to the nearest location? (_It won't be difficult for me to get there again, I just wouldn't want to lose the pearls myself_)

**Comment by snoodledev** [comment]
I believe as long as you don't load your save while it's disabled (ie. load the game), it won't affect your save at all  👍

**Comment by ToxicTrash113** [comment]
Got it! As soon as I get to the PC, I will unsubscribe from all mailings and subscribe to them again

**Comment by ToxicTrash113** [comment]
It helped! Thanks a lot, both gateways are working! I turned everything off to begin with, I left only "The Mast" and "Badlands" (and all the additions to them). Then I connected everything else, but it still works. Thanks again for the help, I really liked the mod and didn't want to lose it.
 _P.S. I don't think that was the problem, but I still note that the "Hoverfly" mod was not installed earlier_
 Thanks again, you helped me out!
